.github/actions/restore-cache/action.yml

```diff
@@ -0,0 +1,130 @@
+name: Restore cached files
+description: Restores backend, vcpkg, and electron-builder from cache
+
+inputs:
+  backend:
+    description: Set to true to cache backend
+    required: false
+    default: 'false'
+
+outputs:
+  backend:
+    description: True if backend was restored from cache
+    value: ${{ steps.backend.outputs.cache-hit }}
+  vcpkg:
+    description: True if vcpkg was restored from cache
+    value: ${{ steps.vcpkg.outputs.cache-hit }}
+  electron-builder:
+    description: True if electron-builder was restored from cache
+    value: ${{ steps.electron-builder.outputs.cache-hit }}
+
+runs:
+  using: 'composite'
+  steps:
+    # The cache folder for electron and electron-builder is different across the three platforms.
+    # Furthermore, the ELECTRON_CACHE and ELECTRON_BUILDER_CACHE environment variables don't work.
+    # So there has to be separate steps for the different platforms to set the proper cache locations,
+    # which were determined by looking at the code here:
+    # https://github.com/develar/app-builder/blob/c87157ae504895a7e9e1329437293846b8e57beb/pkg/download/artifactDownloader.go#L195
+    - name: Set Electron cache paths (macos and linux)
+      if: matrix.os != 'windows-2019'
+      shell: bash
+      run: |
+        if [[ "$RUNNER_OS" == "Linux" ]]; then
+          echo "ELECTRON_CACHE_DIR=$HOME/.cache/electron" >> $GITHUB_ENV
+          echo "ELECTRON_BUILDER_CACHE_DIR=$HOME/.cache/electron-builder" >> $GITHUB_ENV
+        elif [[ "$RUNNER_OS" == "macOS" ]]; then
+          echo "ELECTRON_CACHE_DIR=$HOME/Library/Caches/electron" >> $GITHUB_ENV
+          echo "ELECTRON_BUILDER_CACHE_DIR=$HOME/Library/Caches/electron-builder" >> $GITHUB_ENV
+        fi
+
+    - name: Set Electron cache paths (Windows)
+      if: matrix.os == 'windows-2019'
+      shell: pwsh
+      run: |
+        echo "ELECTRON_CACHE_DIR=$Env:LOCALAPPDATA\electron\Cache" | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf8 -Append
+        echo "ELECTRON_BUILDER_CACHE_DIR=$Env:LOCALAPPDATA\electron-builder\Cache" | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf8 -Append
+
+    # Since Windows doesn't have bash, use separate steps to get the afv-native submodule
```

Comment:
It does on GitHub actions if you specify shell: bash :p

Reply:
Hmm, interesting. Will try.

Reply:
Thanks, that worked. Nice.